lib/WTSI/DNAP/Warehouse/Schema/Result/LighthouseSample.pm

```diff
@@ -63,10 +63,16 @@ Id for this sample provided by the Lighthouse lab
 =head2 cog_uk_id
 
   data_type: 'varchar'
-  is_nullable: 1
+  is_nullable: 0
   size: 255
 
-Consortium-wide id, generated by Sanger on import to LIMS
+=head2 cog_uk_id_unique
+
+  data_type: 'tinyint'
+  default_value: 1
+  is_nullable: 1
+
+A flag to indicate cog_uk_id should be unique. NULL allows reuse of the ID in another row.
```

Comment:
As a side effect it also allows for multiple `cog_uk_id` + `cog_uk_id_unique` values. So you could have:
cog_uk_id: A + cog_uk_id_unique: 1 and cog_uk_id: A + cog_uk_id_unique: 2. Just to be aware if you are assuming a not NULL in `cog_uk_id_unique` guaranties unique `cog_uk_id`.

Reply:
We rarely venture that far into the covid samples territory. When we have to, for a number of reasons it's not easy. Our production code does not go beyond the `sample` table.